src/quantities.hxx

```diff
@@ -177,14 +166,220 @@ auto m_vis(auto &df, const std::string &outputname,
+        return (float)dileptonmet.Pt();
+    };
+    return df.Define(outputname, calculate_pt_tt, {p_1_p4, p_2_p4, met});
+}
+
+/**
+ * @brief function used to calculate the pt of the dilepton + two leading jets +
+ * met system. If the number of jets is less than 2, the quantity is set to 10
+ * instead.
+ *
+ * @param df name of the dataframe
+ * @param outputname name of the new column containing the pt_ttjj value
+ * @param p_1_p4 lorentz vector of the first particle
+ * @param p_2_p4 lorentz vector of the second particle
+ * @param jet_1_p4 lorentz vector of the first jet
+ * @param jet_2_p4 lorentz vector of the second jet
+ * @param met lorentz vector of the met
+ * @return a new dataframe with the new column
+ */
+
+auto pt_ttjj(auto &df, const std::string &outputname, const std::string &p_1_p4,
+             const std::string &p_2_p4, const std::string &jet_1_p4,
+             const std::string &jet_2_p4, const std::string &met) {
+    auto calculate_pt_ttjj = [](ROOT::Math::PtEtaPhiMVector &p_1_p4,
+                                ROOT::Math::PtEtaPhiMVector &p_2_p4,
+                                ROOT::Math::PtEtaPhiMVector &jet_1_p4,
+                                ROOT::Math::PtEtaPhiMVector &jet_2_p4,
+                                ROOT::Math::PtEtaPhiMVector &met) {
+        if (jet_1_p4.pt() < 0.0 || jet_2_p4.pt() < 0.0)
+            return default_float;
+        auto jetlepmet = p_1_p4 + p_2_p4 + met + jet_1_p4 + jet_2_p4;
+        return (float)jetlepmet.Pt();
+    };
+    return df.Define(outputname, calculate_pt_ttjj,
+                     {p_1_p4, p_2_p4, jet_1_p4, jet_2_p4, met});
+}
+
+/**
+ * @brief function used to calculate `mt_tot`. It is defined as
+ \f[
+    m_{T}^{tot} = \sqrt{m_{T}^2(p_{1},E_{T}^{miss}) +
+ m_{T}^2(p_{2},E_{T}^{miss}) + m_{T}^2(p_{1},p_2) } \f] where \f$ m_{T}^2 \f$ is
+ the transverse mass, \f$ p_{1} \f$ and \f$ p_{2} \f$ are the lepton lorentz
+ vectors and \f$ E_{T}^{miss} \f$ is the missing energy.
+ *
+ * @param df name of the dataframe
+ * @param outputname name of the new column containing the mt_tot value
+ * @param p_1_p4 lorentz vector of the first particle
+ * @param p_2_p4 lorentz vector of the second particle
+ * @param met lorentz vector of the met
+ * @return a new dataframe with the new column
+ */
+
+auto mt_tot(auto &df, const std::string &outputname, const std::string &p_1_p4,
+            const std::string &p_2_p4, const std::string &met) {
+    auto calculate_mt_tot = [](ROOT::Math::PtEtaPhiMVector &p_1_p4,
+                               ROOT::Math::PtEtaPhiMVector &p_2_p4,
+                               ROOT::Math::PtEtaPhiMVector &met) {
+        const float mt_1 = vectoroperations::calculateMT(p_1_p4, met);
+        const float mt_2 = vectoroperations::calculateMT(p_2_p4, met);
+        const float mt_mix = vectoroperations::calculateMT(p_1_p4, p_2_p4);
+        return (float)sqrt(mt_1 * mt_1 + mt_2 * mt_2 + mt_mix * mt_mix);
```

Comment:
which doesn't seem to bother you here :)

Reply:
it did !